examples/chat/module/rust/src/proto/chat_grpc.rs

```diff
@@ -43,3 +43,27 @@ pub fn dispatch<T: ChatNode>(node: &mut T, method: &str, req: &[u8], writer: grp
         }
     };
 }
+
+// Client interface trait
+pub trait ChatClient {
+    fn create_room(&self, req: super::chat::CreateRoomRequest) -> grpc::Result<protobuf::well_known_types::Empty>;
+    fn destroy_room(&self, req: super::chat::DestroyRoomRequest) -> grpc::Result<protobuf::well_known_types::Empty>;
+    fn subscribe(&self, req: super::chat::SubscribeRequest) -> grpc::Result<oak::io::Receiver<grpc::GrpcResponse>>;
```

Comment:
I would expect this to return `grpc::Result<oak::io::Receiver<super::chat::Message>>`. I am aware that even the server is currently not as type-safe as it could, so perhaps create an issue to deal with both of these aspects at some point? I think we could have a Rust type to encapsulate `GrpcResponse`, but parametrized by the inner response message type, and implement `To` and `From` `GrpcResponse`.

Reply:
Receiving `chat::Message` objects has the slight wrinkle that you lose the ability to check `grpc_rsp.last`, making it harder to distinguish between no-responses-at-the-moment and method-completed.  However, I guess it might be possible to use `ERR_CHANNEL_CLOSED` from the `Receiver` as a way of handling this; I'll have a think/play.

Reply:
On more of a look, I don't think this can be done without significant changes to `Receiver`.

At the moment `Receiver` is a `struct` with a `ReadHandle` in it, but there is no channel that transports (say) serialized `chat::Message` structures; the only channel available here emits `GrpcResponse` messages.

One possible approach might be to convert `Receiver` to be a `trait`, with some kind of default `impl` for a `ReadHandle`; this case would then use an alternative implementation of the `trait` that did the `GrpcResponse` to `T` / `chat::Message` unwrapping internally.

However, this is all rather beyond the scope of this PR, so I've raised & referenced #592